packages/bloc_test/lib/src/bloc_test.dart

```diff
@@ -231,3 +240,37 @@ Alternatively, consider using Matchers in the expect of the blocTest rather than
   );
   if (errors != null) test.expect(unhandledErrors, test.wrapMatcher(errors()));
 }
+
+String _diff({required dynamic expected, required dynamic actual}) {
+  final buffer = StringBuffer();
+  final differences = diff(expected.toString(), actual.toString());
+  buffer
+    ..writeln('${"=" * 4} diff ${"=" * 40}')
+    ..writeln('')
+    ..writeln(differences.toPrettyString())
+    ..writeln('')
+    ..writeln('${"=" * 4} end diff ${"=" * 36}');
+  return buffer.toString();
+}
+
+extension on List<Diff> {
+  String toPrettyString() {
+    String identical(String str) => '\u001b[90m$str\u001B[0m';
+    String deletion(String str) => '\u001b[31m[-$str-]\u001B[0m';
+    String insertion(String str) => '\u001b[32m{+$str+}\u001B[0m';
+
+    final buffer = StringBuffer();
+    for (final difference in this) {
+      if (difference.operation == DIFF_EQUAL) {
```

Comment:
nit: I believe operation is an enum? If so a switch case may be safer here?

Reply:
Good call, refactored 👍